### PR TITLE
Remove old classes

### PR DESCRIPTION
**Pull request by gogonzo**
This is a feature branch, **DON'T MERGE and DON'T COMMENT**. PR open to have a constant checks. We will start review this PR as a whole in the sprint 4.

### CODE REVIEW

**Review by m7pr — CHANGES_REQUESTED** [review]
just in case to avoid accidental merge

File: R/join_keys-extract.R
Comment (chlebowa) [inline R/join_keys-extract.R:247]: Did you mean this?
```suggestion
    keep_value <- if (is.null(new_x[[ds2]])) list() else new_x[[ds2]]
```

Comment (gogonzo) [reply R/join_keys-extract.R:247]: Thanks, please commit in case you find something

File: R/join_keys-c.R
Comment (chlebowa) [inline R/join_keys-c.R:31]: Is `#nolint` is for the double floor or the period in function name?

We should clean up deviations from our style guide at some point.

File: R/teal_data-get_code.R
Comment (chlebowa) [inline R/teal_data-get_code.R:38]: Adding this method has the unfortunate consequence of `?get_code` being ambiguous.
image

The same goes for `eval_code` in being found `teal` and `teal_code`.

I don't think it can be helped.

File: R/teal_data-get_code.R
Comment (chlebowa) [inline R/teal_data-get_code.R:60]: I think this should be enough.
```suggestion
    parse(text = code, keep.source = TRUE)
```

File: R/verify.R
Comment (chlebowa) [inline R/verify.R:1]:
```suggestion
#' Verify Code Reproducibility
```

File: R/verify.R
Comment (chlebowa) [inline R/verify.R:5]:
```suggestion
#' If objects created by code in the `@code` slot of `x` are `all_equal` to the contents of the `@env` slot,
```

File: vignettes/teal-data-reproducibility.Rmd
Comment (chlebowa) [inline vignettes/teal-data-reproducibility.Rmd:13]: I think the note should be separated from the introduction.
Also, no contractions in official texts.
```suggestion
Reproducibility is a primary function of the `qenv` class, which `teal_data` inherits from. Every data modification in `teal_data` object is performed in an encapsulated environment, separate from the global environment.

It is important to note that the reproducibility of this object is limited only to the data-code relationship. Other aspects such as the reliability of the data source, reproducibility of the R session (including package versions), and creation and use of objects from other environments (e.g. `.GlobalEnvironment`) cannot be verified properly by the `teal_data`.
```

File: vignettes/teal-data.Rmd
Comment (chlebowa) [inline vignettes/teal-data.Rmd:52]:
```suggestion
- `env` - an environment containing data. 
```

File: vignettes/teal-data.Rmd
Comment (chlebowa) [inline vignettes/teal-data.Rmd:53]:
```suggestion
- `code` - a string containing code to reproduce `env` (details in [teal_data reproducibility ](teal-data-reproducibility.html)).
```

File: vignettes/teal-data.Rmd
Comment (chlebowa) [inline vignettes/teal-data.Rmd:59]:
```suggestion
The primary function of `teal_data` is to provide reproducibility of data.
We recommend to initialize empty `teal_data`, which marks object as _verified_, and create data sets with code evaluated in the object, using `within` or `eval_code`.
Read more in [teal_data Reproducibility](teal-data-reproducibility.html).
```

File: tests/testthat/test-get_code.R
Comment (chlebowa) [inline tests/testthat/test-get_code.R:3]:
```suggestion
testthat::test_that("get_code with datanames handles empty @code slot", {
```

File: R/deprecated.R
Comment (gogonzo) [reply R/deprecated.R:3]: adding suggestion to handle this here
```suggestion
#' @description `r lifecycle::badge("deprecated")`\br
```

Comment (chlebowa) [reply R/deprecated.R:3]: *`\cr`
You can also do
```
#' @description
#' <badge>
#' 
#' <text>
```

File: R/deprecated.R
Comment (gogonzo) [inline R/deprecated.R:3]:
```suggestion
#' @description `r lifecycle::badge("deprecated")`\cr
```

File: R/join_keys-c.R
Comment (averissimo) [reply R/join_keys-c.R:31]: It's for the "\_\_parents\_\_" name. We can also replace the attribute name to "parents" without underscores

```suggestion
      attr(out, "__parents__") <- .merge_parents(.x, .y) # nolint: object_name_linter.
```

File: R/teal_data-get_code.R
Comment (gogonzo) [reply R/teal_data-get_code.R:38]: I'm worried that there is no other way

**Comment by averissimo** [comment]
Does it makes sense to cleanup linter tags? Comment on PR #212 against this branch

**Comment by gogonzo** [comment]
> Does it makes sense to cleanup linter tags? Comment on PR https://github.com/insightsengineering/teal.data/pull/212 against this branch

Perhaps, can we decide after merge? #212 targets `main` so it probably has to wait

File: R/default_cdisc_join_keys.R
Comment (averissimo) [inline R/default_cdisc_join_keys.R:68]: Function is also defined in `data_label.R`. One of them should be removed.

Should it be called `get_cdisc_primary_keys` ?

File: R/verify.R
Comment (chlebowa) [reply R/verify.R:60]: They should for the arguments that are dispatched on. Is there anything missing?

Comment (averissimo) [reply R/verify.R:60]: This would be the change.

```suggestion
setMethod("verify", signature = c("teal_data"), definition = function(x) {
```

File: R/utils.R
Comment (averissimo) [reply R/utils.R:37]: Here it is

image

File: R/default_cdisc_join_keys.R
Comment (gogonzo) [reply R/default_cdisc_join_keys.R:68]: I'm removing this at all and use `default_cdisc_join_keys["ADSL", "ADSL"]`. It is used just in few files in the whole organization

Comment (averissimo) [reply R/default_cdisc_join_keys.R]: Great!! One occurence left on https://github.com/insightsengineering/user-guide/blob/477b72fbae1b172fbde30e89d5c0704ed80301f7/content/teal/using_study_data.md?plain=1#L59

What's the process for that repo, can we just change it now or should we wait till the release?

File: R/utils.R
Comment (chlebowa) [reply R/utils.R:37]: Cools :+1:

File: R/default_cdisc_join_keys.R
Comment (gogonzo) [reply R/default_cdisc_join_keys.R]: Whaaat, what is it! I'll probably remove it. It's redundant I guess

Comment (gogonzo) [reply R/default_cdisc_join_keys.R]: This repo has been update last time 2 years ago. Probably has failures already. It is not a blocker.

**Review by vedhav — APPROVED** [review]

**Pull request merged**
